systems/snl-eldorado/system.py

```diff
@@ -0,0 +1,730 @@
+                "mpi": {"buildable": False},
+                "libfabric": {
+                    "externals": [
+                        {"spec": "libfabric@2.1", "prefix": "/opt/cray/libfabric/2.1"}
+                    ],
+                    "buildable": False,
+                },
+            }
+        }
+
+        selections["packages"] |= self.rocm_config()["packages"]
+
+        selections["packages"] |= self.mpi_config()["packages"]
+
+        if self.spec.satisfies("compiler=cce"):
+            selections["packages"] |= {
+                "cray-libsci": {
+                    "externals": [
+                        {
+                            "spec": "cray-libsci@23.05.1.4%cce",
+                            "prefix": "/opt/cray/pe/libsci/23.05.1.4/cray/12.0/x86_64/",
+                        }
+                    ]
+                }
+            }
+        elif self.spec.satisfies("compiler=gcc"):
+            selections["packages"] |= {
+                "cray-libsci": {
+                    "externals": [
+                        {
+                            "spec": "cray-libsci@23.05.1.4%gcc",
+                            "prefix": "/opt/cray/pe/libsci/23.05.1.4/gnu/10.3/x86_64/",
+                        }
+                    ]
+                }
+            }
+
+        selections["packages"] |= self.compiler_weighting_cfg()["packages"]
+
+        return selections
+
+    def compiler_weighting_cfg(self):
+        compiler = self.spec.variants["compiler"][0]
+
+        if compiler == "cce":
+            return {"packages": {"all": {"require": [{"one_of": ["%cce", "%gcc"]}]}}}
+        elif compiler == "gcc":
+            return {"packages": {}}
+        elif compiler == "rocmcc":
+            return {"packages": {"all": {"require": [{"one_of": ["%rocmcc", "%gcc"]}]}}}
+        else:
+            raise ValueError(f"Unexpected value for compiler: {compiler}")
+
+    def compute_compilers_section(self):
+        selections = {
+            "compilers": [
+                {
+                    "compiler": {
+                        "spec": "gcc@12.2.0",
+                        "paths": {
+                            "cc": "/opt/cray/pe/gcc/12.2.0/bin/gcc",
+                            "cxx": "/opt/cray/pe/gcc/12.2.0/bin/g++",
+                            "f77": "/opt/cray/pe/gcc/12.2.0/bin/gfortran",
+                            "fc": "/opt/cray/pe/gcc/12.2.0/bin/gfortran",
+                        },
+                        "flags": {},
+                        "operating_system": "rhel8",
+                        "target": "x86_64",
+                        "modules": [],
+                        "environment": {},
+                        "extra_rpaths": [],
+                    }
+                }
+            ]
+        }
+        if self.spec.satisfies("compiler=cce") or self.spec.satisfies(
+            "compiler=rocmcc"
+        ):
+            selections["compilers"] += self.rocm_cce_compiler_cfg()["compilers"]
```

Comment:
The specification for this changed: it looks like https://github.com/LLNL/benchpark/blob/develop/systems/llnl-elcapitan/system.py is a close match (uses all the same compilers). If you have questions about that, let me know.